src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+                              calls: Traversable[Call],
+                              backend: Backend): Future[Unit] = {
+
+    val action = for {
+
+      workflowExecutionInsert <- dataAccess.workflowExecutionsAutoInc +=
+        new WorkflowExecution(
+          workflowInfo.workflowId.toString,
+          WorkflowSubmitted.toString,
+          new Date().toTimestamp)
+
+      _ <- dataAccess.workflowExecutionAuxesAutoInc += new WorkflowExecutionAux(
+        workflowExecutionInsert.workflowExecutionId.get,
+        workflowInfo.wdlSource,
+        workflowInfo.wdlJson)
+
+      symbolInsert <- dataAccess.symbolsAutoInc ++=
+        workflowInputs.toSeq map { symbol =>
+          new Symbol(
+            workflowExecutionInsert.workflowExecutionId.get,
+            symbol.key.scope,
+            symbol.key.name,
+            symbol.key.iteration,
+            if (symbol.isInput) IoInput else IoOutput,
+            symbol.wdlType.toWdlString,
+            symbol.wdlValue.map(wdlValueToString).orNull)
+        }
+
+      // NOTE: Don't use DBIO.seq for **transforming** sequences
+      // - DBIO.seq(mySeq: _*) runs *any* items in sequence, but converts Seq[ DBIOAction[_] ] to DBIOAction[ Unit ]
+      // - DBIO.sequence(mySeq) converts Seq[ DBIOAction[R] ] to DBIOAction[ Seq[R] ]
+      // - DBIO.fold(mySeq, init) converts Seq[ DBIOAction[R] ] to DBIOAction[R]
+
+      _ <- DBIO.sequence(calls.toSeq map {
+        call => for {
+          executionInsert <- dataAccess.executionsAutoInc +=
+            new Execution(
+              workflowExecutionInsert.workflowExecutionId.get,
+              call.taskFqn,
+              ExecutionStatus.NotStarted.toString)
+
+          _ <- backend match {
+            case _: LocalBackend =>
+              dataAccess.localJobsAutoInc +=
+                new LocalJob(
+                  executionInsert.executionId.get,
+                  -1,
+                  -1)
+            case null =>
+              throw new NotImplementedError("Backend is null")
+            case unknown =>
+              throw new NotImplementedError("Unknown backend: " + backend.getClass)
+          }
+        } yield ()
+      })
+
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  override def getWorkflowState(workflowId: WorkflowId): Future[Option[WorkflowState]] = {
+
+    val action = for {
+
+      workflowExecutionResultOption <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.headOption
+
+      workflowState = workflowExecutionResultOption map { workflowExecutionResult =>
+        WorkflowState.fromString(workflowExecutionResult.status)
+      }
+
+    } yield workflowState
+
+    runTransaction(action)
+  }
+
+  override def getExecutionStatuses(workflowId: WorkflowId): Future[Map[FullyQualifiedName, CallStatus]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
```

Comment:
All of these `head` calls are intentionally here to throw an `Exception`, getting then wrapped in a `Failure` and returned. Alternatively, it's easy enough to use `headOption` to return a `Future[None]` instead.

I'm not sure how/why the current code would end up with a bad uuid for this particular call, so I assumed it was an error. I haven't looked, but I assume the server will eventually need other methods with `Future[Option[_]]` that it can use for `rejectEmptyResponse` / `404`.